experimental/slo/slo_middleware.go

```diff
@@ -0,0 +1,47 @@
+package slo
+
+import (
+	"errors"
+	"net/http"
+	"syscall"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/httpclient"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+var duration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+	Namespace: "grafana",
+	Name:      "plugin_external_requests_duration",
+	Help:      "Duration of requests to external services",
+}, []string{"plugin", "error_source"})
+
+// Middleware captures duration of requests to external services and the source of errors
+func Middleware(plugin string) httpclient.Middleware {
```

Comment:
Nit. Suggest extracting the plugin type as here https://github.com/grafana/grafana/blob/aae6b5312a7412a3fb8db160d8a34c2730a50415/pkg/infra/httpclient/httpclientprovider/datasource_metrics_middleware.go#L75 to make the Middleware more generic.

Reply:
updated.  please review again